source/mir/math/common.d

```diff
@@ -217,11 +217,83 @@ version(LDC)
+    ` else static if (T.mant_dig == double.mant_dig) return gcc.builtins.__builtin_`~fun~`(x, y);`~
+    ` else static if (T.mant_dig == real.mant_dig) return gcc.builtins.__builtin_`~fun~`l(x, y);`~
+    ` else static assert(0);`;
+
+    ///
+    T sqrt(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`sqrt`); }
+    ///
+    T sin(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`sin`); }
+    ///
+    T cos(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`cos`); }
+    ///
+    T pow(T)(in T x, in T power) if (isFloatingPoint!T) { alias y = power; mixin(mixinGCCBuiltin2!`pow`); }
+    ///
+    T powi(T)(in T x, int power) if (isFloatingPoint!T) { alias y = power; mixin(mixinGCCBuiltin2!`powi`); }
+    ///
+    T exp(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`exp`); }
+    ///
+    T log(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`log`); }
+    ///
+    T fabs(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`fabs`); }
+    ///
+    T floor(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`floor`); }
+    ///
+    T exp2(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`exp2`); }
+    ///
+    T log10(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`log10`); }
+    ///
+    T log2(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`log2`); }
+    ///
+    T ceil(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`ceil`); }
+    ///
+    T trunc(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`trunc`); }
+    ///
+    T rint(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`rint`); }
+    ///
+    T nearbyint(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`nearbyint`); }
+    ///
+    T copysign(T)(in T mag, in T sgn) if (isFloatingPoint!T) { alias y = sgn; mixin(mixinGCCBuiltin2!`copysign`); }
+    ///
+    T round(T)(in T x) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin!`round`); }
+    ///
+    T fmuladd(T)(in T a, in T b, in T c) if (isFloatingPoint!T)
+    {
+        static if (T.mant_dig == float.mant_dig)
+            return gcc.builtins.__builtin_fmaf(a, b, c);
+        else static if (T.mant_dig == double.mant_dig)
+            return gcc.builtins.__builtin_fma(a, b, c);
+        else static if (T.mant_dig == real.mant_dig)
+            return gcc.builtins.__builtin_fmal(a, b, c);
+        else
+            static assert(0);
+    }
+    ///
+    T fmin(T)(in T x, in T y) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin2!`fmin`); }
+    ///
+    T fmax(T)(in T x, in T y) if (isFloatingPoint!T) { mixin(mixinGCCBuiltin2!`fmax`); }
+}
 else
 {
     static import std.math;
+    static import core.math;
     ///
-    T sqrt(T)(in T x) if (isFloatingPoint!T) { return std.math.sqrt(x); }
+    T sqrt(T)(in T x) if (isFloatingPoint!T) { return core.math.sqrt(x); }
     ///
     T sin(T)(in T x) if (isFloatingPoint!T) { return std.math.sin(x); }
```

Comment:
`std` is used here. Could you please add a small unittest, one function should be ok, and GDC to travis with this PR?

Reply:
Sure

Reply:
Uninttest added. I was wrong about travis: GDC is being tested. But "latest gdc" on travis has frontend version 2.068.2 so most of mir can't compile.